api/billingd/service/service.go

```diff
@@ -755,6 +771,55 @@ func (s *Service) DeleteCustomer(ctx context.Context, req *pb.DeleteCustomerRequ
 	return &pb.DeleteCustomerResponse{}, nil
 }
 
+func (s *Service) GetCustomerUsage(
+	ctx context.Context,
+	req *pb.GetCustomerUsageRequest,
+) (*pb.GetCustomerUsageResponse, error) {
+	doc, err := s.getCustomer(ctx, "_id", req.Key)
+	if err != nil {
+		return nil, err
+	}
+	usage := make(map[string]*pb.Usage)
+	for k, u := range doc.DailyUsage {
+		if product, ok := s.products[k]; ok {
+			// Get reported usage over the current invoice period
+			free, err := s.getPeriodUsageItem(u.FreeItemID)
+			if err != nil {
+				return nil, err
+			}
+			paid, err := s.getPeriodUsageItem(u.PaidItemID)
+			if err != nil {
+				return nil, err
+			}
+			total := (free.TotalUsage + paid.TotalUsage) * product.UnitSize
+			// Add current day unreported usage
+			total += u.Total
+			usage[k] = getUsage(product, total, doc.InvoicePeriod)
+		}
+	}
+	return &pb.GetCustomerUsageResponse{
+		Usage: usage,
+	}, nil
+}
+
+func (s *Service) getPeriodUsageItem(id string) (sum *stripe.UsageRecordSummary, err error) {
+	params := &stripe.UsageRecordSummaryListParams{
+		SubscriptionItem: stripe.String(id),
+	}
+	params.Filters.AddFilter("limit", "", "1")
```

Comment:
Only return the most recent (current) invoice period.

Reply:
Not sure about code around this, but is the `OrderBy` criteria `Desc` by default? Asking just in case isn't the oldest.

Reply:
Yep, "reverse-chronological" order. Docs [here](https://stripe.com/docs/api/usage_records/subscription_item_summary_list).